src/platform/backends/hyperv_api/hcs_plan9_mount_handler.cpp

```diff
@@ -0,0 +1,194 @@
+
+            mpu::make_target_dir(session, leading, missing);
+            mpu::set_owner_for(session, leading, missing, default_uid, default_gid);
+        }
+
+        // fmt::format("sudo mount -t 9p {} {} -o trans=virtio,version=9p2000.L,msize=536870912",
+        // tag, target));
+        constexpr std::string_view mount_command_fmtstr =
+            "sudo mount -t 9p -o trans=virtio,version=9p2000.L,port={} {} {}";
+
+        const auto& add_settings = std::get<hcs::HcsAddPlan9ShareParameters>(req.settings);
+        const auto mount_command =
+            fmt::format(mount_command_fmtstr, add_settings.port, add_settings.access_name, target);
+
+        auto mount_command_result = session.exec(mount_command);
+
+        if (mount_command_result.exit_code() == 0)
+        {
+            mpl::info(log_category,
+                      "Successfully mounted 9P share `{}` to VM `{}`",
+                      req,
+                      vm->get_name());
+        }
+        else
+        {
+            mpl::error(log_category,
+                       "stdout: {} stderr: {}",
+                       mount_command_result.read_std_output(),
+                       mount_command_result.read_std_error());
+            throw std::runtime_error{"Failed to mount the Plan9 share"};
+        }
+    }
+    catch (...)
+    {
+        const auto req = [this] {
+            hcs::HcsRemovePlan9ShareParameters params{};
+            params.name = mpu::make_uuid(target).remove("-").left(30).prepend('m').toStdString();
+            params.access_name =
+                mpu::make_uuid(target).remove("-").left(30).prepend('m').toStdString();
+            return hcs::HcsRequest{hcs::HcsResourcePath::Plan9Shares(),
+                                   hcs::HcsRequestType::Remove(),
+                                   params};
+        }();
+        if (!HCS().modify_compute_system(handle, req))
+        {
+            // TODO: Warn here
+        }
+    }
+}
+void Plan9MountHandler::deactivate_impl(bool force)
+{
+    SSHSession session{vm->ssh_hostname(), vm->ssh_port(), vm->ssh_username(), *ssh_key_provider};
+    constexpr std::string_view umount_command_fmtstr =
+        "mountpoint -q {0}; then sudo umount {0}; else true; fi";
+    const auto umount_command = fmt::format(umount_command_fmtstr, target);
+
+    if (!(session.exec(umount_command).exit_code() == 0))
+    {
+        // TODO: Include output?
+        mpl::warn(log_category, "Plan9 share unmount failed.");
+
+        if (!force)
+        {
+            return;
+        }
+    }
+
+    const auto req = [this] {
+        hcs::HcsRemovePlan9ShareParameters params{};
+        params.name = mpu::make_uuid(target).remove("-").left(30).prepend('m').toStdString();
+        params.access_name = mpu::make_uuid(target).remove("-").left(30).prepend('m').toStdString();
+        return hcs::HcsRequest{hcs::HcsResourcePath::Plan9Shares(),
+                               hcs::HcsRequestType::Remove(),
```

Comment:
This is already in the hcs namespace, so better remove those qualifications, no?